printer/PrintHandler.js

```diff
@@ -57,39 +58,46 @@ function determinePrinterForJob() {
 
 setInterval(async () => {
   const data = await readMessageFromSqs(params, sqs);
-  if (!data) {
-    return;
-  }
+  if (!data) return;
+  const fromDiscord = (data.Body.fileURL) ? true : false;
+  const printer = determinePrinterForJob();
+
+  const { fileNo, copies = 1, pageRanges = 'NA'} = data.Body;
 
-  const { fileNo, copies, pageRanges } = data.Body;
   const pages = pageRanges === 'NA' ? '' : '-P ' + pageRanges;
-  const path = `/tmp/${fileNo}.pdf`;
+  let filePath = `/tmp/${fileNo}.pdf`;
 
-  const paramers = {
-    Bucket: PRINTING_BUCKET_NAME,
-    Key: `folder/${fileNo}.pdf`,
-  };
+  if (!fromDiscord) {
+    const paramers = {
+      Bucket: PRINTING_BUCKET_NAME,
+      Key: `folder/${fileNo}.pdf`,
+    };
 
-  s3.getObject(paramers, (err, dataFromS3) => {
-    if (err) console.error(err);
-    fs.writeFileSync(path, dataFromS3.Body, 'binary');
-    const printer = determinePrinterForJob();
-    exec(
-      `lp -n ${copies} ${pages} -o sides=one-sided -d ` +
-      `HP-LaserJet-p2015dn-${printer} ${path}`,
-      (error, stdout, stderr) => {
-        if (error) throw error;
-        if (stderr) throw stderr;
-        exec(`rm ${path}`, () => { });
-        const deleteParams = {
-          QueueUrl: queueUrl,
-          ReceiptHandle: data.ReceiptHandle,
-        };
-        deleteFile(fileNo);
-        sqs.deleteMessage(deleteParams, (err) => {
-          if (err) throw err;
-        });
+    s3.getObject(paramers, (err, dataFromS3) => {
+      if (err) console.error(err);
+      fs.writeFileSync(filePath, dataFromS3.Body, 'binary');
+      const deleteParams = {
+        QueueUrl: queueUrl,
+        ReceiptHandle: data.ReceiptHandle,
+      };
+      deleteFile(fileNo);
+      sqs.deleteMessage(deleteParams, (err) => {
+        if (err) throw err;
       });
-  });
+    }); 
+  } else {
+    const fileDownloaded = await downloadFileFromURL(data.Body.fileURL);
```

Comment:
add one here as well, a reader who doesnt know much about the file would be lost aka mee